### PR TITLE
Feature/client/e3 create expense form

### PR DESCRIPTION
**Pull request by Atsuki02**
## Describe your changes

- Create create-expense-form
- Modify create/update-expense api

## Issue

#98 

## URL

http://localhost:3000/expenses/create

## Screenshot (if you made the UI)

https://github.com/HAK2024/fairy-share/assets/132407687/e8e75440-612b-4a87-90fe-9df47a60e00f

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings and no errors

### CODE REVIEW

File: client/app/expenses/hooks/useCreateExpense.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const houseId = useGetHouseInfo().houseId
+
+  const { mutate, isPending } = useCreateExpenseMutation()
+
+  const form = useForm<CreateExpenseSchema>({
+    resolver: createExpenseResolver,
+    // TODO: Set empty to fee if I figured out a way
+    defaultValues: {
+      itemName: '',
+      fee: 0,
```
Comment (Atsuki02) [inline client/app/expenses/hooks/useCreateExpense.ts:18]: It's hard to set null as default value bcause of type, so I'll fix it when I have time in the future

**Review by kanatagu — COMMENTED** [review]
Could you delete `client/app/expenses/type/index.ts` as it's an empty file?

File: client/app/_components/ui/field/DateField.tsx [hunk 2]
```diff
@@ -17,11 +17,13 @@ import {
 export type DateFieldProps<TFormValues extends FieldValues> = {
   control: Control<TFormValues>
   name: Path<TFormValues>
+  type: 'task' | 'expense'
```
[the remark this answers is not in the document]
Comment (Atsuki02) [reply client/app/_components/ui/field/DateField.tsx]: Fixed!
https://github.com/HAK2024/fairy-share/pull/107/commits/eef809ce966b6be285b072c0e6bb5e6e9e0f83a3

**Comment by Atsuki02** [comment]
> Could you delete `client/app/expenses/type/index.ts` as it's an empty file?

Fixed
https://github.com/HAK2024/fairy-share/pull/107/commits/f73d7f5a66cb0141bc9d85267c403f4aa4096ba4

File: client/app/expenses/components/CreateExpenseForm.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+              <FormItem>
+                <FormLabel isRequired>Expenses&nbsp;($)</FormLabel>
+                <FormControl>
+                  <Input
+                    type='number'
+                    placeholder='0.00'
+                    min='0.01'
+                    step='0.01'
+                    {...field}
+                  />
```
[the remark this answers is not in the document]
Comment (Atsuki02) [reply client/app/expenses/components/CreateExpenseForm.tsx:48]: Fixed!
https://github.com/HAK2024/fairy-share/pull/107/commits/cda4a6b295e0d7d684ee8d8ffee3bf5a42992372

File: client/app/expenses/hooks/api/useCreateExpenseMutation.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useMutation } from '@tanstack/react-query'
+import { createExpenseApi } from '@/expenses/api'
+import { ExpenseSchema } from '@/expenses/schema'
```
[the remark this answers is not in the document]
Comment (Atsuki02) [reply client/app/expenses/hooks/api/useCreateExpenseMutation.ts]: Fixed!
https://github.com/HAK2024/fairy-share/pull/107/commits/f4d557ac4ce8f253cad0f8d80ba102070400ea5c

File: client/app/expenses/hooks/useCreateExpense.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      ...data,
+      houseId,
+    }
+    mutate(formattedData, {
+      onSuccess: () => {
+        toast({
+          variant: 'success',
+          title: 'New expense has been created successfully',
+        })
+        form.reset()
```
[the remark this answers is not in the document]
Comment (Atsuki02) [reply client/app/expenses/hooks/useCreateExpense.ts]: Fixed!
https://github.com/HAK2024/fairy-share/pull/107/commits/3d5833123e6c4b78ca148109b6727a32b3d50e8d

File: client/app/expenses/create/page.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+'use client'
+
+import { FormContainer } from '@/_components/layout'
+import { Heading } from '@/_components/ui'
+import { CreateExpenseForm } from '../components'
+
+export default function ExpenseCreatePage() {
```
[the remark this answers is not in the document]
Comment (Atsuki02) [reply client/app/expenses/create/page.tsx]: Fixed!
https://github.com/HAK2024/fairy-share/pull/107/commits/f4d557ac4ce8f253cad0f8d80ba102070400ea5c

File: client/app/expenses/hooks/useCreateExpense.ts (hunk 1 above)
Comment (Atsuki02) [reply client/app/expenses/hooks/useCreateExpense.ts:20]: Okay! I'll fix it in the future PR!

**Pull request merged**